src/test/users-query.test.ts

```diff
@@ -0,0 +1,101 @@
+import * as request from 'supertest';
+import { expect } from 'chai';
+import { getRepository } from 'typeorm';
+import { User } from '../entity/User';
+import * as jwt from 'jsonwebtoken';
+
+const query = `
+query Users($data: UsersQuery!) {
+  users(data: $data) {
+    users {
+      name
+      id
+      email
+    }
+    page
+    totalPage
+    hasPreviousPage
+    hasNextPage
+  }
+}`;
+
+async function graphqlPost(variables: any, token: string) {
+  return request('localhost:4001')
+    .post('/')
+    .send({ query, variables: { data: variables } })
+    .set({ Authorization: token });
+}
+
+const validToken: string = jwt.sign({ id: 1 }, process.env.JWT_KEY, {
+  expiresIn: process.env.ACCESS_TOKEN_EXPIRES_IN,
+});
+describe('user-query test', function () {
+  it('should give an error if token is invalid', async function () {
+    const variables = {};
+
+    const response = await graphqlPost(variables, 'invalid token');
+
+    const expectedResponse = { message: 'Invalid token', code: 401 };
+    expect(response.body.errors[0].message).to.equal(expectedResponse.message);
+    expect(response.body.errors[0].extensions.exception.code).to.equal(expectedResponse.code);
+  });
+
+  it('should give an error if page is negative', async function () {
+    const variables = { page: -1 };
+
+    const response = await graphqlPost(variables, validToken);
+
+    const expectedResponse = { message: 'page must not be negative', code: 400 };
+    expect(response.body.errors[0].message).to.equal(expectedResponse.message);
+    expect(response.body.errors[0].extensions.exception.code).to.equal(expectedResponse.code);
+  });
+
+  it('should give an error if page is higher than the last page', async function () {
+    const userRepository = getRepository(User);
+    const variables = { page: 10, limit: 25 };
+    const count = await userRepository.count();
+    const totalPage = Math.floor(count / variables.limit);
+
+    const response = await graphqlPost(variables, validToken);
+
+    const expectedResponse = { message: `page must be between 1 and ${totalPage}`, code: 400 };
+    expect(response.body.errors[0].message).to.equal(expectedResponse.message);
+    expect(response.body.errors[0].extensions.exception.code).to.equal(expectedResponse.code);
+  });
```

Comment:
Esse caso é que eu acho que poderia retornar uma lista vazia. Se o front tiver fazendo uma paginação infiita ou algo do tipo, a lista vazia pode ser uma boa condição de parada pra eles. Se for jogado um erro, fica difícil ele diferenciar entre um erro mesmo ou apenas o fim dos usuários. Faz sentido?